src/bbs2gh/Commands/MigrateRepoCommand.cs

```diff
@@ -2,141 +2,239 @@
+        Description += Environment.NewLine;
+        Description += "Note: Expects GH_PAT env variable or --github-pat option to be set.";
 
-        public MigrateRepoCommand(
-            OctoLogger log,
-            GithubApiFactory githubApiFactory,
-            EnvironmentVariableProvider environmentVariableProvider
-        ) : base("migrate-repo")
+        // Arguments to generate a new archive
+        var bbsServerUrl = new Option<string>("--bbs-server-url")
         {
-            _log = log;
-            _githubApiFactory = githubApiFactory;
-            _environmentVariableProvider = environmentVariableProvider;
+            IsRequired = false,
+            Description = "The full URL of the Bitbucket Server/Data Center to migrate from."
+        };
 
-            Description = "Import a Bitbucket Server archive to GitHub.";
-            Description += Environment.NewLine;
-            Description += "Note: Expects GH_PAT env variable or --github-pat option to be set.";
+        var bbsProject = new Option<string>("--bbs-project")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket project to import; defaults to '*' (all projects)."
+        };
 
-            var archiveUrl = new Option<string>("--archive-url")
-            {
-                IsRequired = true,
-                Description = "URL used to download Bitbucket Server migration archive."
-            };
-            var githubOrg = new Option<string>("--github-org")
-            {
-                IsRequired = true
-            };
-            var githubRepo = new Option<string>("--github-repo")
-            {
-                IsRequired = true
-            };
-            var githubPat = new Option<string>("--github-pat")
-            {
-                IsRequired = false
-            };
-            var wait = new Option("--wait")
-            {
-                Description = "Synchronously waits for the repo migration to finish."
-            };
-            var verbose = new Option("--verbose")
-            {
-                IsRequired = false
-            };
+        var bbsRepo = new Option<string>("--bbs-repo")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket repository to import; defaults to '*' (all repositories)."
+        };
+
+        var bbsUsername = new Option<string>("--bbs-username")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket username of a user with project admin privileges."
+        };
+
+        var bbsPassword = new Option<string>("--bbs-password")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket password of the user specified by --bbs-username."
+        };
+
+        // Arguments to import an existing archive
+        var archiveUrl = new Option<string>("--archive-url")
+        {
+            IsRequired = false,
+            Description = "URL used to download Bitbucket Server migration archive."
```

Comment:
```suggestion
            Description = "URL used to download Bitbucket Server migration archive. Only needed if you want to manually retrieve the archive from BBS instead of letting this CLI do that for you."
```